From 8f53babccb149fa22d84c7a42343f1a9eda56be4 Mon Sep 17 00:00:00 2001
From: Jian He <jianhe@apache.org>
Date: Mon, 13 Apr 2015 10:53:34 -0700
Subject: [PATCH 1377/2748] YARN-3382. Some of UserMetricsInfo metrics are
 incorrectly set to root queue metrics.
 Contributed by Rohit Agarwal (cherry picked from
 commit 944a16579fdb54c1c420162ef830ef2ac5451c69)

(cherry picked from commit 32b01263d3c9b5469f1b7ff9f21170823a8bf10c)

Change-Id: I66a5fb858bd3390d80f7fa1b862ac3daae7fddc8
---
 .../webapp/dao/UserMetricsInfo.java                |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/dao/UserMetricsInfo.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/dao/UserMetricsInfo.java
index 73a83d7..bfa5bd2 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/dao/UserMetricsInfo.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/dao/UserMetricsInfo.java
@@ -63,11 +63,11 @@ public UserMetricsInfo(final ResourceManager rm, final String user) {
       this.userMetricsAvailable = true;
 
       this.appsSubmitted = userMetrics.getAppsSubmitted();
-      this.appsCompleted = metrics.getAppsCompleted();
-      this.appsPending = metrics.getAppsPending();
-      this.appsRunning = metrics.getAppsRunning();
-      this.appsFailed = metrics.getAppsFailed();
-      this.appsKilled = metrics.getAppsKilled();
+      this.appsCompleted = userMetrics.getAppsCompleted();
+      this.appsPending = userMetrics.getAppsPending();
+      this.appsRunning = userMetrics.getAppsRunning();
+      this.appsFailed = userMetrics.getAppsFailed();
+      this.appsKilled = userMetrics.getAppsKilled();
 
       this.runningContainers = userMetrics.getAllocatedContainers();
       this.pendingContainers = userMetrics.getPendingContainers();
-- 
1.7.9.5

